### ISSUE TITLE
[bug] V2 Breaking change - options require name/version instead of just name

### ISSUE THREAD

**Issue by John-LittleBearLabs** [conan-io/conan#13417]
### Environment details

* Operating System+version: Fedora 36
* Compiler+version: g++ 12.2.1
* Conan version: 2.0.0
* Python version: 3.10.10

### Steps to reproduce

1. Make a conanfile.txt with:
```
[requires]
boost/1.81.0
[generators]
CMakeDeps
[options]
# V1 allows...
boost:header_only=True
# V2 requires boost/1.81.0:header_only=True
```
2. conan install .

### Logs

```
======== Input profiles ========
Profile host:
[settings]
arch=x86_64
build_type=Release
compiler=gcc
compiler.cppstd=gnu17
compiler.libcxx=libstdc++11
compiler.version=12
os=Linux
[conf]


Profile build:
[settings]
arch=x86_64
build_type=Release
compiler=gcc
compiler.cppstd=gnu17
compiler.libcxx=libstdc++11
compiler.version=12
os=Linux
[conf]


ERROR: Error while parsing [options] in conanfile
Options should be specified as 'pkg:option=value'
```

**Comment by pacemdei** [comment]
In contrast to the situation with the above mentioned profiles, a conditional dependency  in the `configure()` method in `conanfile.py`, does not result in a warning, but simply ignores the option for the dependency i.e.
```python
configure(self):
    self.options["dependency"].option  = foo
```
is ignored. Is this by design? I could not find any documentation about it.

On the other hand using wildcards seem to work
i.e.
```python
configure(self):
    self.options["dependency/*"].option  = foo
```
works.

## Environment details
conan:      2.0.1
compiler: gcc 12.2.1
os :            Fedora 37
python:    3.11.2

**Comment by memsharded** [comment]
Hi @John-LittleBearLabs 

Thanks @pacemdei for the hints.

Yes, Conan 2.0 requires the name to be a full exact match, like ``zlib/1.2.13`` or to be a pattern, like ``zlib*``. But only the partial ``zlib`` name is no longer a match. This is expected and necessary to eliminate some ambiguity and issues that happened in 1.X.

The recommend syntax that will be jointly compatible in 1.X and 2.0 would be: ``self.options["zlib*"].some_option``, and when possible in 2.0-only, it would be better to do ``self.options["zlib/*"].some_option``.

**Comment by memsharded** [comment]
We will be trying to restore the ``self.options["mydep"].myoption`` syntax in https://github.com/conan-io/conan/pull/13467, for 2.0.3.

**Comment by memsharded** [comment]
https://github.com/conan-io/conan/pull/13467 merged, it will be in 2.0.3
